### PR TITLE
qa: Extract rpc_timewait as test param

### PR DESCRIPTION
**Pull request by maflcko**
Also increase it for wallet_dump and wallet_groups

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Timeout in `wallet_dump` (observed on https://bitcoinperf.com/):

```
[1mwallet_dump.py[0m failed, Duration: 145 s[1mstdout:
[0m2018-08-01T16:16:03.257000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_test_runner_20180801_121146/wallet_dump_80
2018-08-01T16:17:28.529000Z TestFramework (ERROR): JSONRPC error
Traceback (most recent call last):
 File "/tmp/bench-master-2018-08-01-1s9v29g9/bitcoin/test/functional/test_framework/authproxy.py", line 153, in _get_response
   http_response = self.__conn.getresponse()
 File "/usr/lib/python3.5/http/client.py", line 1198, in getresponse
   response.begin()
 File "/usr/lib/python3.5/http/client.py", line 297, in begin
   version, status, reason = self._read_status()
 File "/usr/lib/python3.5/http/client.py", line 258, in _read_status
   line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
 File "/usr/lib/python3.5/socket.py", line 576, in readinto
   return self._sock.recv_into(b)
socket.timeout: timed outDuring handling of the above exception, another exception occurred:Traceback (most recent call last):
 File "/tmp/bench-master-2018-08-01-1s9v29g9/bitcoin/test/functional/test_framework/test_framework.py", line 160, in main
   self.run_test()
 File "/tmp/bench-master-2018-08-01-1s9v29g9/bitcoin/test/functional/wallet_dump.py", line 153, in run_test
   self.nodes[0].importwallet(wallet_unenc_dump)
 File "/tmp/bench-master-2018-08-01-1s9v29g9/bitcoin/test/functional/test_framework/coverage.py", line 47, in __call__
   return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
 File "/tmp/bench-master-2018-08-01-1s9v29g9/bitcoin/test/functional/test_framework/authproxy.py", line 136, in __call__
   response = self._request('POST', self.__url.path, postdata.encode('utf-8'))
 File "/tmp/bench-master-2018-08-01-1s9v29g9/bitcoin/test/functional/test_framework/authproxy.py", line 107, in _request
   return self._get_response()
 File "/tmp/bench-master-2018-08-01-1s9v29g9/bitcoin/test/functional/test_framework/authproxy.py", line 160, in _get_response
   self.__conn.timeout)})
test_framework.authproxy.JSONRPCException: 'importwallet' RPC took longer than 60.000000 seconds. Consider using larger timeout for calls that take longer to return. (-344)
```

Timeout on `wallet_groups` (observed on macOS):

```
wallet_groups.py failed, Duration: 129 s
stdout:
2018-08-01T17:31:47.460000Z TestFramework (INFO): Initializing test directory /var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/bitcoin_test_runner_20180801_172522/wallet_groups_46
2018-08-01T17:32:55.584000Z TestFramework (ERROR): JSONRPC error
Traceback (most recent call last):
  File "/Users/travis/build/MarcoFalke/btc_nightly/bitcoin/test/functional/test_framework/authproxy.py", line 153, in _get_response
    http_response = self.__conn.getresponse()
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1331, in getresponse
    response.begin()
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 297, in begin
    version, status, reason = self._read_status()
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 258, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/socket.py", line 586, in readinto
    return self._sock.recv_into(b)
socket.timeout: timed out
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/Users/travis/build/MarcoFalke/btc_nightly/bitcoin/test/functional/test_framework/test_framework.py", line 160, in main
    self.run_test()
  File "/Users/travis/build/MarcoFalke/btc_nightly/bitcoin/test/functional/wallet_groups.py", line 90, in run_test
    assert self.nodes[2].sendtoaddress(address=addr2[0], amount=5)
  File "/Users/travis/build/MarcoFalke/btc_nightly/bitcoin/test/functional/test_framework/coverage.py", line 47, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/Users/travis/build/MarcoFalke/btc_nightly/bitcoin/test/functional/test_framework/authproxy.py", line 136, in __call__
    response = self._request('POST', self.__url.path, postdata.encode('utf-8'))
  File "/Users/travis/build/MarcoFalke/btc_nightly/bitcoin/test/functional/test_framework/authproxy.py", line 107, in _request
    return self._get_response()
  File "/Users/travis/build/MarcoFalke/btc_nightly/bitcoin/test/functional/test_framework/authproxy.py", line 160, in _get_response
    self.__conn.timeout)})
test_framework.authproxy.JSONRPCException: 'sendtoaddress' RPC took longer than 60.000000 seconds. Consider using larger timeout for calls that take longer to return. (-344)

**Comment by jamesob** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/13837/commits/fa5b440971a0dfdd64c1b86748a573fcd7dc65d3. Thanks!

File: test/functional/test_framework/test_node.py [hunk 1]
```diff
@@ -56,17 +56,13 @@ class TestNode():
     To make things easier for the test writer, any unrecognised messages will
     be dispatched to the RPC connection."""
 
-    def __init__(self, i, datadir, rpchost, timewait, bitcoind, bitcoin_cli, mocktime, coverage_dir, extra_conf=None, extra_args=None, use_cli=False):
+    def __init__(self, i, datadir, *, rpchost, timewait, bitcoind, bitcoin_cli, mocktime, coverage_dir, extra_conf=None, extra_args=None, use_cli=False):
```
Comment (jamesob) [inline test/functional/test_framework/test_node.py]: TIL: https://www.python.org/dev/peps/pep-3102/

**Comment by laanwj** [comment]
utACK fa5b440971a0dfdd64c1b86748a573fcd7dc65d3

**Pull request merged**